### ISSUE TITLE
Import annotations?

### ISSUE THREAD

**Issue by tmaiaroto** [pdfcpu/pdfcpu#365]
Is it possible to import annotations into an existing PDF? 

I have an XFDF (or FDF) file and want to import that into an existing PDF. I imagine I could even parse the XML version to get all of the attributes and create new annotations with the various functions I've seen in the source code...but maybe there's another more direct way even.

I don't know if there's other projects for this entirely, but I think this project has some functions that sound like they may work (and I know golang vs other projects in languages I don't know). Maybe it's something not exposed to the CLI and/or documented and I'm just missing it when skimming through the source.

**Comment by kaiceo** [comment]
Why not try https://pkg.go.dev/github.com/pdfcpu/pdfcpu@v0.3.12/pkg/api#AddAnnotationsMapFile

**Comment by hhrutter** [comment]
What is your usecase?
Are you trying to fill an Acroform?
Form support is being worked on.

**Comment by tmaiaroto** [comment]
I'm trying to take annotations exported from: https://pdfjs.express as XFDF and import them into a PDF. While this product does have an API to generate a PDF with the annotations merged, it has limited support for what I'm trying to do (which is modify/add annotations that number others like footnotes for print along with a separate page of comments and info which have the corresponding number).

So my thinking is that I can take the XFDF file, add/edit XML and then apply those annotations to a PDF. That may require converting  XFDF to FDF maybe, I'm not sure. Or, it could just mean looping the XFDF contents and taking all of the attributes from the XML and using them as arguments for various functions to add annotations (sounds like it'd be slower maybe, but maybe not, certainly more involved).

So no need for form filling and such, just taking the annotations and trying to get them merged in.

**Comment by hhrutter** [comment]
XFDF contains form data and like I said pdfcpu form processing is still being worked on.

**Comment by tmaiaroto** [comment]
It also contains annotations though. Can any form related stuff be skipped over? Though sure, forms would be cool too. I just was after the annotations and if there was any kind of existing functionality to take something like:

`<circle name="D5244D6B32F9E84B63A17E116BD661ED" style="solid" creationdate="D:20210727134801-07&apos;00&apos;" width="1.0" flags="print" date="D:20210727134823-07&apos;00&apos;" color="#ff0000" opacity="1.0" page="0" subject="Circle" rect="307.0,657.7055,381.4621,677.4637" title="test"/>`

and call an `addAnnotation()` function of some kind giving it arguments from the XML node attributes.

Here's an example XFDF with a few different types of annotations for reference if helpful.

```
<?xml version="1.0" encoding="UTF-8"?><xfdf xml:space="preserve" xmlns="http://ns.adobe.com/xfdf/"><f href="sample.pdf"/>
    <annots>
        <circle name="D5244D6B32F9E84B63A17E116BD661ED" style="solid" creationdate="D:20210727134801-07&apos;00&apos;" width="1.0" flags="print" date="D:20210727134823-07&apos;00&apos;" color="#ff0000" opacity="1.0" page="0" subject="Circle" rect="307.0,657.7055,381.4621,677.4637" title="test"/>
        <square name="FB1464D8204DD03A1A6058DE87787F86" style="solid" creationdate="D:20210727134825-07&apos;00&apos;" width="1.0" flags="print" date="D:20210727134825-07&apos;00&apos;" color="#ff0000" opacity="1.0" page="0" subject="Square" rect="138.9808,685.5459,203.1628,697.7103" title="test"/>
        <highlight name="EDD2A658992D09A8B53ADD92AAAA131C" creationdate="D:20210727134835-07&apos;00&apos;" flags="print" date="D:20210727134835-07&apos;00&apos;" color="#ffff00" opacity="1.0" page="0" subject="Highlight Text" rect="70.075,685.108,93.875,697.838" title="test" coords="72.03,697.838,90.92,697.838,72.03,686.108,90.92,686.108"/>
        <line IT="LineArrow" rect="381.9042,660.1579,429.1301,674.1577" opacity="1.0" end="427.1301,667.2067" start="383.9042,666.9625" head="None" page="0" subject="Arrow" flags="print" tail="ClosedArrow" width="1.0" color="#ff0000" interior-color="#ff0000" style="solid" caption="no" date="D:20210727134839-07&apos;00&apos;" creationdate="D:20210727134839-07&apos;00&apos;" title="test" name="ED467FDF7B0400F87CC55971C4562CC6"/>
        <line IT="LineArrow" rect="202.407,664.4741,419.85,699.4509" opacity="1.0" end="417.85,666.4741" start="204.407,693.5818" head="ClosedArrow" page="0" subject="Arrow" flags="print" tail="None" width="1.0" color="#ff0000" interior-color="#ff0000" style="solid" caption="no" date="D:20210727134842-07&apos;00&apos;" creationdate="D:20210727134842-07&apos;00&apos;" title="test" name="8079504123751C249D7EC991CCA2C668"/>
    </annots>
</xfdf>
```

**Comment by hhrutter** [comment]
You're right. [The spec](https://www.iso.org/obp/ui/#iso:std:iso:19444:-1:ed-2:v1:en) states:
```
XML Forms Data Format (XFDF) is a format for representing forms data and annotations in a PDF document. 
```

Unfortunately importing annotations from XFDF is not available right now.
Sounds like a useful feature though.
